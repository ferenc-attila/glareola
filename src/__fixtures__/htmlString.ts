export const htmlString: string = `<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=iso-8859-2">
</head>
<body>
<div class="Keret">
    <div class="TonusosDoboz">
        <div class="Kozep">
            <table border="0" cellpadding="6" cellspacing="0" width="100%">
                <tr class="formtable">
                    <td width="1">&nbsp;</td>
                    <td colspan="2">
                        <table border="0" cellpadding="3" cellspacing="0" class="formtable" width="100%">
                            <tr>
                                <td width="125"><strong>Időpont:</strong></td>
                                <td>1935. december 28.</td>
                                <td rowspan="8" valign="top" align="right" width="16">&nbsp;</td>
                            </tr>
                            <tr>
                                <td width="125"><strong>Faj:</strong></td>
                                <td><a href="javascript:loginAlert()" class="Bordo">Mezei veréb</a> - <em>Passer
                                    montanus</em></td>
                            </tr>
                            <tr>
                                <td width="125"><strong>Példány, kor, ivar:</strong></td>
                                <td>1,&nbsp;&nbsp;ad.</td>
                            </tr>
                            <tr>
                                <td width="125"><strong>Település, megye:</strong></td>
                                <td><span class="text">
									Eger, Heves								</span></td>
                            </tr>

                            <tr>
                                <td width="125"><strong>Terület:</strong></td>
                                <td>Nagy-Eged</td>
                            </tr>
                            <tr>
                                <td width="125"><strong>Megfigyelő(k):</strong></td>
                                <td>John Doe, Jane Doe</td>
                            </tr>
                            <tr>
                                <td width="125"><strong>Feltöltő:</strong></td>
                                <td><a class="Bordo" href="http://birding.hu/useradatlap/0000">Jack Doe</a></td>
                            </tr>

                            <tr>
                                <td width="125"><strong>Megjegyzés:</strong></td>
                                <td>
                                    <div style="word-wrap:break-word;width:500px">Testing data</div>
                                </td>
                            </tr>

                            <tr>
                                <script type="text/javascript">

                                    var gMapLat = '47.927998';
                                    var gMapLong = '20.410659';


                                    function callGoogleMaps(lat, long) {
                                        window.open('https://www.google.com/maps/place/' + lat + 'N,' + long + 'E', '2_blank');
                                    }

                                    function showAddress(address) {
                                        console.log(address);
                                        a = address.replace(/, Magyaror.*/g, '');
                                        console.log(a);
                                        $.getJSON('https://dev.utcakereso.hu/terkep.php/geocoder/get/' + a, function (data) {
                                                lon = parseFloat(data.lon).toFixed(6);
                                                lat = parseFloat(data.lat).toFixed(6);

                                                if (isNaN(lon)) lon = 19.56;
                                                if (isNaN(lat)) lat = 47.16;

                                                $('#longitude').val(lat);
                                                $('#lattitude').val(lon);

                                                gMapLat = lat;
                                                gMapLong = lon;

                                                newLatLng = new L.LatLng(lat, lon);
                                                marker.setLatLng(newLatLng).bindTooltip(lon + ',' + lat);
                                                ;
                                                mymap.setView(newLatLng);
                                            }
                                        );
                                    }

                                    $('document').ready(function () {
                                    });
                                </script>


                                <td class="szoveg">
                                    <strong>GPS-koordináták: </strong>
                                </td>
                                <td>
                                    <span style="white-space:nowrap;display:inline;">É:&nbsp;<input type="text"
                                                                                                    name="longitude"
                                                                                                    id="longitude"
                                                                                                    readonly
                                                                                                    style="width:80px;"
                                                                                                    value="47.927998"></span>
                                    <span style="white-space:nowrap;display:inline;">K:&nbsp;<input type="text"
                                                                                                    name="lattitude"
                                                                                                    id="lattitude"
                                                                                                    readonly
                                                                                                    style="width:80px;"
                                                                                                    value="20.410659"></span>
                                    <input type="button" value="google"
                                           onclick="javascript: callGoogleMaps(gMapLat, gMapLong);">
                                </td>
                            </tr>
                            <tr>
                                <td>&nbsp;</td>
                                <td>&nbsp;</td>
                            </tr>
                            <tr>
                                <td colspan="2">
                                    <div id="map_canvas" style="z-index:0;width:395px;height:220px;"></div>
                                    <link rel="stylesheet" href="https://unpkg.com/leaflet@1.4.0/dist/leaflet.css"
                                          integrity="sha512-puBpdR0798OZvTTbP4A8Ix/l+A4dHDD0DGqYW6RQ+9jxkRFclaxxQb/SJAWZfWAkuyeQUytO7+7N4QKrDh+drA=="
                                          crossorigin=""/>
                                    <script src="https://unpkg.com/leaflet@1.4.0/dist/leaflet.js"
                                            integrity="sha512-QVftwZFqvtRNi0ZyCtsznlKSWOStnDORoefr1enyq5mVL4tmKB3S/EnC3rRJcxCPavG10IcrVGSmPh6Qw5lwrg=="
                                            crossorigin=""></script>

                                    <script>
                                        var mymap = L.map('map_canvas').setView([47.927998, 20.410659], 13);
                                        L.tileLayer('https://utcakereso.hu/tile/osm/{z}/{x}/{y}.png?access_token={accessToken}', {
                                                attribution: 'Map data &copy; <a href="https://www.openstreetmap.org/">OpenStreetMap</a>, <a href="https://creativecommons.org/licenses/by-sa/2.0/">CC-BY-SA</a>,<a href="https://utcakereso.hu/">Utcakereso.hu</a>',
                                                maxZoom: 18,
                                                id: 'mapbox.streets',
                                                accessToken: 'your.mapbox.access.token'
                                            }
                                        ).addTo(mymap);


                                        var marker = L.marker([47.927998, 20.410659],
                                            {
                                                draggable: false
                                            }
                                        ).addTo(mymap);
                                    </script>

                                </td>
                                <td width="240" valign="top" align="center"><a class="thickbox" href="../../assets/images/glareola_pratincola.png"><img src="../../assets/images/glareola_pratincola.png" border="0"></a>						<br>
							<a href="http://www.birding.hu/kepertekeles/00000" class="Bordo">Értékelés:</a> 5/5

						 <a href="http://www.birding.hu/kepertekeles/00000" class="Bordo">Kommentek száma:</a> 0
					</td>
                            </tr>

                        </table>
                        <br>
                    </td>

                </tr>
            </table>
            <script type="text/javascript">
                function loginAlert() {
                    alert('A tartalom megtekintéséhez be kell jelentkezni!');
                }
            </script>
        </div>
    </div>
</div>
</body>
</html>`;
