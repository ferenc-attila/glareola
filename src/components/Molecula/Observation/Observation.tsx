import React, { useEffect, useState } from 'react';
import { StyleSheet, Text, View, Modal, Image } from 'react-native';

import i18n from '../../../localization';
import { IBirdingHuData } from '../../../types/interfaces';
import { GlareolaButton } from '../../Atomic/GlareolaButton';

export const Observation = (observationData: IBirdingHuData) => {
    const [modalVisible, setModalVisible] = useState(false);
    const [modalContent, setModalContent] = useState('none');

    useEffect(() => {
        if (!modalVisible) {
            setModalContent('none');
        }
    }, [modalVisible]);

    useEffect(() => {
        if (modalContent !== 'none') {
            setModalVisible(true);
        }
    }, [modalContent]);

    return (
        <>
            <View style={styles.container}>
                <View style={styles.observationContainer}>
                    <Text style={styles.observationHeader}>{observationData.speciesHun}</Text>
                    <Text style={styles.observationBody}>{observationData.locality}</Text>
                    <Text style={styles.observationBody}>{observationData.county}</Text>
                    <Text style={styles.observationBody}>{observationData.area}</Text>
                    <Text style={styles.observationBody}>{observationData.date}</Text>
                    <Text style={styles.observationBody}>{observationData.individuals}</Text>
                    <Text style={styles.observationBody}>{observationData.observers.join(', ')}</Text>
                    {observationData.notes && <Text style={styles.observationBody}>{observationData.notes}</Text>}
                    {observationData.imageLink && (
                        <GlareolaButton
                            onPress={() => setModalContent('image')}
                            iconName='camera'
                            size={24}
                            color='white'
                            accessibilityLabel={i18n.t('BUTTON_IMAGE_LABEL')}
                        />
                    )}
                </View>
            </View>
            <Modal
                animationType='slide'
                transparent
                visible={modalVisible}
                onRequestClose={() => {
                    setModalVisible(!modalVisible);
                }}
            >
                <View style={styles.modalContainer}>
                    <View style={styles.modal}>
                        {modalContent === 'image' && observationData.imageLink && (
                            <Image
                                style={styles.image}
                                source={{ uri: observationData.imageLink }}
                                accessibilityRole='image'
                                accessibilityLabel={`${i18n.t('OBSERVATION_IMAGE_DESCRIPTION')} ${
                                    observationData.speciesSci
                                }`}
                            />
                        )}
                        <GlareolaButton
                            onPress={() => setModalVisible(!modalVisible)}
                            iconName='backward'
                            size={24}
                            color='white'
                            accessibilityLabel={i18n.t('BUTTON_CLOSE_LABEL')}
                        />
                    </View>
                </View>
            </Modal>
        </>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#2e3a24',
        width: '100%',
        alignItems: 'center',
        justifyContent: 'center',
    },
    observationContainer: {
        borderRadius: 10,
        backgroundColor: '#595947',
        width: '90%',
        alignItems: 'center',
        justifyContent: 'center',
        padding: 10,
        margin: 10,
    },
    observationHeader: {
        fontSize: 20,
        fontWeight: 'bold',
        color: '#fff',
        padding: 5,
    },
    observationBody: {
        fontSize: 16,
        color: '#fff',
        textAlign: 'center',
    },
    modalContainer: {
        flex: 1,
        backgroundColor: '#2e3a24',
        alignItems: 'center',
        justifyContent: 'center',
    },
    modal: {
        backgroundColor: '#a9a996',
        padding: 10,
        margin: 10,
        width: '90%',
        height: '90%',
        alignItems: 'center',
        justifyContent: 'center',
    },
    image: {
        margin: 10,
        width: '100%',
        height: '90%',
        resizeMode: 'contain',
    },
});
