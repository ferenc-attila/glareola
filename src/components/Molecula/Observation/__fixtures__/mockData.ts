import { SOURCES } from '../../../../constants';
import { IBirdingHuData } from '../../../../types/interfaces';

export const mockBirdingHuData: IBirdingHuData = {
    source: SOURCES.BIRDING_HU,
    webId: 1,
    speciesHun: 'Székicsér',
    speciesSci: 'Glareola pratincola',
    individuals: 1,
    date: '2021-05-01',
    locality: 'Budapest',
    county: 'Budapest',
    area: 'Parlament',
    latitude: 47.497913,
    longitude: 19.040236,
    observers: ['John Doe', 'Jane Doe'],
    uploader: 'John Doe',
    notes: 'Some comment',
    imageLink: 'assets/images/glareola_pratincola.png',
};
