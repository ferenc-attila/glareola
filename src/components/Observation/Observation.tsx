import React, { useEffect, useState } from 'react';
import { StyleSheet, Text, View, Modal, Image, Pressable } from 'react-native';
import MapView, { Marker } from 'react-native-maps';
import { IBirdingHuData } from '../../types/types';

export const Observation = (observationData: IBirdingHuData) => {
    const [modalVisible, setModalVisible] = useState(false);
    const [modalContent, setModalContent] = useState('none');

    useEffect(() => {
        if (!modalVisible) {
            setModalContent('none');
        }
    }, [modalVisible]);

    useEffect(() => {
        if (modalContent !== 'none') {
            setModalVisible(true);
        }
    }, [modalContent]);

    return (
        <>
            <View style={styles.container}>
                <View style={styles.observationContainer}>
                    <Text style={styles.observationHeader}>
                        {observationData.speciesHun}
                    </Text>
                    <Text style={styles.observationBody}>
                        {observationData.locality}
                    </Text>
                    <Text style={styles.observationBody}>
                        {observationData.county}
                    </Text>
                    <Text style={styles.observationBody}>
                        {observationData.area}
                    </Text>
                    <Text style={styles.observationBody}>
                        {observationData.date}
                    </Text>
                    <Text style={styles.observationBody}>
                        {observationData.individuals}
                    </Text>
                    <Text style={styles.observationBody}>
                        {observationData.observers.join(', ')}
                    </Text>
                    {observationData.notes
                        && <Text style={styles.observationBody}>
                        {observationData.notes}
                    </Text>}
                    <Pressable onPress={() => setModalContent('map')}>
                        <Text style={styles.button}> Show on map </Text>
                    </Pressable>
                    {observationData.imageLink
                        && <Pressable onPress={() => setModalContent('image')}>
                            <Text style={styles.button}> Show image </Text>
                        </Pressable>}
                </View>
            </View>
            <Modal
                animationType="slide"
                transparent={true}
                visible={modalVisible}
                onRequestClose={() => {
                    setModalVisible(!modalVisible);
                }}
            >
                <View style={styles.modalContainer}>
                    <View style={styles.modal}>
                        {modalContent == 'map'
                            && <MapView
                                style={styles.map}
                                initialRegion={{
                                    latitude: observationData.latitude,
                                    longitude: observationData.longitude,
                                    latitudeDelta: 0.0922,
                                    longitudeDelta: 0.0421,
                                }}
                            >
                                <Marker
                                    coordinate={{
                                        latitude: observationData.latitude,
                                        longitude: observationData.longitude,
                                    }}
                                    title={observationData.speciesSci}
                                    description={observationData.locality}
                                />
                            </MapView>}
                        {modalContent == 'image'
                            && observationData.imageLink
                            && <Image
                                style={styles.image}
                                source={{uri: observationData.imageLink}}
                            />}
                        <Pressable onPress={() => setModalVisible(!modalVisible)}>
                            <Text style={styles.button}>Close</Text>
                        </Pressable>
                    </View>
                </View>
            </Modal>
        </>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#2e3a24',
        width: '100%',
        alignItems: 'center',
        justifyContent: 'center',
    },
    observationContainer: {
        borderRadius: 10,
        backgroundColor: '#595947',
        width: '90%',
        alignItems: 'center',
        justifyContent: 'center',
        padding: 10,
        margin: 10,
    },
    observationHeader: {
        fontSize: 20,
        fontWeight: 'bold',
        color: '#fff',
        padding: 5,
    },
    observationBody: {
        fontSize: 16,
        color: '#fff',
        textAlign: 'center',
    },
    button: {
        padding: 10,
        margin: 10,
        borderRadius: 5,
        backgroundColor: '#4d9460',
        color: '#ffffff',
        fontSize: 15,
        fontWeight: 'bold'
    },
    modalContainer: {
        flex: 1,
        backgroundColor: '#2e3a24',
        alignItems: 'center',
        justifyContent: 'center',
    },
    modal: {
        backgroundColor: '#a9a996',
        padding: 10,
        margin: 10,
        width: '90%',
        height: '90%',
        alignItems: 'center',
        justifyContent: 'center',
    },
    map: {
        width: '100%',
        height: '90%',
        margin: 10,
    },
    image: {
        margin: 10,
        width: '100%',
        height: '90%',
        resizeMode: 'contain',
    },
});
