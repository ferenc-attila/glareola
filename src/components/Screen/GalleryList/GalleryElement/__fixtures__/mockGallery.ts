import { SOURCES } from '../../../../../constants';
import { IBirdingHuGalleryElement } from '../../../../../types/interfaces';

export const mockBirdingHuGalleryElement: IBirdingHuGalleryElement = {
    id: '12345678',
    author: 'John Doe',
    source: SOURCES.BIRDING_HU,
    speciesHun: 'Székicsér',
    speciesSci: 'Glareola pratincola',
    date: '2021-05-01',
    locality: 'Budapest',
    area: 'Parlament',
    notes: 'Some comment',
    smallImageLink: '../src/assets/images/glareola_icon.png',
    detailsUrl: '../../__fixtures__/birdingHuHtmlStrings.ts',
    cameraType: 'Canon EOS 5D Mark IV',
    cameraSettings: ['f/5.6', '1/1000', 'ISO 400'],
    imageLink: '../src/assets/images/glareola_pratincola.png',
};
