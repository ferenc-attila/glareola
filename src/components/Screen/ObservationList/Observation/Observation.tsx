import React, { useEffect, useState } from 'react';
import { StyleSheet, Text, View, Modal, Image, Linking, Platform } from 'react-native';

import { ICON_NAMES } from '../../../../constants';
import i18n from '../../../../localization';
import { COLORS, CONTAINER_STYLES, SPACING, TEXT_STYLES } from '../../../../styles';
import { IBirdingHuData } from '../../../../types/interfaces';
import { GlareolaButton } from '../../../Atomic/GlareolaButton';

export const Observation = (observationData: IBirdingHuData) => {
    const [modalVisible, setModalVisible] = useState(false);
    const [modalContent, setModalContent] = useState('none');

    useEffect(() => {
        if (!modalVisible) {
            setModalContent('none');
        }
    }, [modalVisible]);

    useEffect(() => {
        if (modalContent !== 'none') {
            setModalVisible(true);
        }
    }, [modalContent]);

    const openObservationDetails = async (url: string) => {
        await Linking.openURL(url);
    };

    const openMapApplication = async () => {
        const scheme = Platform.select({ ios: 'maps://0,0?q=', android: 'geo:0,0?q=' });
        const latLng = `${observationData.latitude},${observationData.longitude}`;
        const label = observationData.speciesHun ?? observationData.speciesSci;
        const url = Platform.select({
            ios: `${scheme}${label}@${latLng}`,
            android: `${scheme}${latLng}(${label})`,
        });
        if (typeof url === 'string') {
            await Linking.openURL(url);
        }
    };

    return (
        <>
            <View style={styles.observationContainer}>
                <Text style={styles.sectionTitleText}>{observationData.speciesHun}</Text>
                <Text style={styles.commonText}>{observationData.locality}</Text>
                <Text style={styles.commonText}>{observationData.county}</Text>
                <Text style={styles.commonText}>{observationData.area}</Text>
                <Text style={styles.commonText}>{observationData.date}</Text>
                <Text style={styles.commonText}>{observationData.individuals}</Text>
                {observationData.notes ? (
                    <>
                        <Text style={styles.commonText}>{observationData.observers.join(', ')}</Text>
                        <Text style={styles.lastCommonText}>{observationData.notes}</Text>
                    </>
                ) : (
                    <Text style={styles.lastCommonText}>{observationData.observers.join(', ')}</Text>
                )}
                <View style={styles.observationButtonContainer}>
                    <GlareolaButton
                        onPress={() => openObservationDetails(observationData.details)}
                        iconName={ICON_NAMES.INFO}
                        accessibilityLabel={i18n.t('BUTTON_OBSERVATION_INFO_LABEL')}
                    />
                    <GlareolaButton
                        onPress={openMapApplication}
                        iconName={ICON_NAMES.MAP}
                        accessibilityLabel={i18n.t('BUTTON_OBSERVATION_MAP_LABEL')}
                    />
                    {observationData.imageLink ? (
                        <GlareolaButton
                            onPress={() => setModalContent('image')}
                            iconName={ICON_NAMES.IMAGE}
                            accessibilityLabel={i18n.t('BUTTON_IMAGE_LABEL')}
                        />
                    ) : null}
                </View>
            </View>
            <Modal style={{ padding: SPACING.LARGE }} animationType='fade' transparent visible={modalVisible}>
                <View style={styles.modal}>
                    {modalContent === 'image' && !!observationData.imageLink ? (
                        <Image
                            style={styles.image}
                            source={{ uri: observationData.imageLink }}
                            accessibilityRole='image'
                            accessibilityLabel={`${i18n.t('OBSERVATION_IMAGE_DESCRIPTION')} ${
                                observationData.speciesSci
                            }`}
                        />
                    ) : null}
                    <GlareolaButton
                        onPress={() => setModalVisible(!modalVisible)}
                        iconName={ICON_NAMES.BACK}
                        accessibilityLabel={i18n.t('BUTTON_CLOSE_LABEL')}
                    />
                </View>
            </Modal>
        </>
    );
};

const styles = StyleSheet.create({
    observationContainer: {
        ...CONTAINER_STYLES.COMMON_CONTAINER,
    },
    observationButtonContainer: {
        ...CONTAINER_STYLES.BUTTON_CONTAINER,
    },
    modal: {
        backgroundColor: COLORS.BACKGROUND.DATA,
        padding: SPACING.LARGE,
        marginTop: SPACING.LARGE,
        width: '90%',
        height: '90%',
        alignItems: 'center',
        justifyContent: 'space-around',
        alignSelf: 'center',
        borderRadius: SPACING.SMALL,
    },
    image: {
        margin: SPACING.MEDIUM,
        width: '100%',
        height: '90%',
        resizeMode: 'contain',
    },
    sectionTitleText: {
        ...TEXT_STYLES.SECTION_TITLE,
    },
    commonText: {
        ...TEXT_STYLES.COMMON,
    },
    lastCommonText: {
        ...TEXT_STYLES.COMMON,
        marginBottom: SPACING.MEDIUM,
    },
});
