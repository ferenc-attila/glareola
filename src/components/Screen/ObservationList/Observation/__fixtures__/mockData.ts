import { SOURCES } from '../../../../../constants';
import { IBirdingHuData } from '../../../../../types/interfaces';

export const mockBirdingHuData: IBirdingHuData = {
    source: SOURCES.BIRDING_HU,
    details: 'http://birding.hu/megfigyeles/adatlap/12345678',
    webId: 1,
    speciesHun: 'Székicsér',
    speciesSci: 'Glareola pratincola',
    individuals: 1,
    date: '2021-05-01',
    locality: 'Budapest',
    county: 'Budapest',
    area: 'Parlament',
    latitude: 47.497913,
    longitude: 19.040236,
    observers: ['John Doe', 'Jane Doe'],
    uploader: 'John Doe',
    notes: 'Some comment',
    imageLink: '../src/assets/images/glareola_pratincola.png',
};
