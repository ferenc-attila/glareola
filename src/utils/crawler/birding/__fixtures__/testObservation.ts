import { IBirdingHuData } from '../../../../types/interfaces';

export const testObservation: IBirdingHuData = {
    source: 'birding.hu',
    webId: 12345678,
    date: '1935. december 28.',
    speciesHun: 'Mezei ver√©b',
    speciesSci: 'Passer montanus',
    individuals: 1,
    locality: 'Eger',
    county: 'Heves',
    area: 'Nagy-Eged',
    observers: ['John Doe', 'Jane Doe'],
    uploader: 'Jack Doe',
    notes: 'Testing data',
    latitude: 47.927998,
    longitude: 20.410659,
    imageLink: '../src/assets/images/glareola_pratincola.png',
};
