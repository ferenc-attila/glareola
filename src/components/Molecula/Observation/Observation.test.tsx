import { fireEvent, render } from '@testing-library/react-native/build/pure';
import { screen } from '@testing-library/react-native/build/screen';

import { Observation } from './Observation';
import { mockBirdingHuData } from './__fixtures__/mockData';

describe('Observation', () => {
    test('should render correctly', () => {
        const container = render(<Observation {...mockBirdingHuData} />);
        expect(container.toJSON()).toMatchSnapshot();
    });

    test('should open image modal if show image button pressed', async () => {
        render(<Observation {...mockBirdingHuData} />);
        const imageButton = screen.getByLabelText('Show image button');
        fireEvent.press(imageButton);
        expect(screen.getByLabelText('Image created during observation: Glareola pratincola')).toBeTruthy();
    });

    test('should close image modal if close button pressed', async () => {
        render(<Observation {...mockBirdingHuData} />);
        const imageButton = screen.getByLabelText('Show image button');
        fireEvent.press(imageButton);
        const closeButton = screen.getByLabelText('Close button');
        fireEvent.press(closeButton);
        expect(screen.queryByLabelText('Image created during observation: Glareola pratincola')).toBeFalsy();
    });
});
