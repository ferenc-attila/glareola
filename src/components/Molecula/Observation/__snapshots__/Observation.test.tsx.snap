// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Observation should render correctly 1`] = `
[
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#2e3a24",
        "flex": 1,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#595947",
          "borderRadius": 10,
          "justifyContent": "center",
          "margin": 10,
          "padding": 10,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 20,
            "fontWeight": "bold",
            "padding": 5,
          }
        }
      >
        Székicsér
      </Text>
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Budapest
      </Text>
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Budapest
      </Text>
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Parlament
      </Text>
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        2021-05-01
      </Text>
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        1
      </Text>
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        John Doe, Jane Doe
      </Text>
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Some comment
      </Text>
      <View
        style={
          {
            "backgroundColor": "#4d9460",
            "borderRadius": 5,
            "color": "#ffffff",
            "fontSize": 15,
            "fontWeight": "bold",
            "margin": 10,
            "padding": 10,
          }
        }
      >
        <View
          accessibilityLabel="Show image button"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Text />
        </View>
      </View>
    </View>
  </View>,
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  />,
]
`;
